packages/tc-api-s3-document-store/patch.js

```diff
@@ -11,8 +11,13 @@ const s3Patch = async ({ s3Instance, bucketName, type, code, body }) => {
 		code,
 	});
 	const changedProperties = Object.keys(body).filter(
-		key => body[key] !== existingBody[key],
+		// check that at least one has a value to avoid e.g. null !== undefined causing
+		// an unnecessary write (the api will send nulls sometimes, but the json object will
+		// contain undefineds)
+		key =>
+			(body[key] || existingBody[key]) && body[key] !== existingBody[key],
```

Comment:
Does the && belong to the !== existing key or the OR?  Some extra brackets would help my understanding.

Reply:
I agree, but our linting rules will convert it back to this though (somethig like 'no extraneous brackets')